src/java.desktop/windows/native/libawt/windows/awt_PrintJob.cpp

```diff
@@ -1050,16 +1269,51 @@ Java_sun_awt_windows_WPrinterJob_validatePaper(JNIEnv *env, jobject self,
 
     jmethodID setSizeID = env->GetMethodID(paperClass,
                                         SETSIZE_STR, SETSIZE_SIG);
-    JNI_CHECK_NULL_GOTO(setSizeID, "no setSize method", done);
+    if (setSizeID == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "no setSize method");
+        /* Free any resources allocated */
+        if (privateDC == TRUE) {
+            if (printDC != NULL) {
+                /* In this case we know that this DC has no GDI objects to free */
+                ::DeleteDC(printDC);
+            }
+            if (hDevMode != NULL) {
+                ::GlobalFree(hDevMode);
+            }
+            if (hDevNames != NULL) {
+                ::GlobalFree(hDevNames);
+            }
+        }
+        return;
+    }
 
     jmethodID setImageableID = env->GetMethodID(paperClass,
                                         SETIMAGEABLE_STR, SETIMAGEABLE_SIG);
-    JNI_CHECK_NULL_GOTO(setImageableID, "no setImageable method", done);
+    if (setImageableID == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "no setImageable method");
+        /* Free any resources allocated */
+        if (privateDC == TRUE) {
+            if (printDC != NULL) {
+                /* In this case we know that this DC has no GDI objects to free */
+                ::DeleteDC(printDC);
+            }
+            if (hDevMode != NULL) {
+                ::GlobalFree(hDevMode);
+            }
+            if (hDevNames != NULL) {
+                ::GlobalFree(hDevNames);
+            }
+        }
+        return;
+    }
+
 
     env->CallVoidMethod(newPaper, setSizeID, paperWidth, paperHeight);
     env->CallVoidMethod(newPaper, setImageableID, ix, iy, iw, ih);
 
-done:
+
```

Comment:
```suggestion
```

Reply:
I recommend deleting this line, since it does not make sense to have two consecutive blank lines here.